### ISSUE TITLE
Structured Logging

### ISSUE THREAD

**Issue by adam-26** [lazojs/lazo#284]
Hi,

Currently the lazo logger formats all log entries as a string, then passes the string value to each sink. Have you considered having the lazo logger structure each log entry as an object literal containing the log data, and then let each sink format the data as it requires?

Example: 
```javascript
var logEntry = {
    timestamp: '2015-01-01T00:00:00.000Z',
    level: 'DEBUG',
    requestId: '123',
    tags: [],
    message: 'This is a message',
    error: [Object object]
};

// Custom log entry
LAZO.logger.debug(logEntry);

// If the first argument passed in an instance of the Error class, it could be assigned to an error property
// to allow for custom handling in sinks.
LAZO.logger.error(err, 'This is a %s message', 'error');
```

This would allow developers to provide more to sinks in a structured manner so they they do not have to attempt to parse strings if they require the data in a different structure. Developers could also pass object literals directly to the lazo logger functions.

Maintaining backwards compatibility with the current logging format doesn't look like it would be a problem. Is this something you would accept in a PR?

Adam.

**Comment by jstrimpel** [comment]
Yes please provided that backwards compatibility can be maintain as you noted. Thanks, Jason
